api/v1/logstorage_types.go

```diff
@@ -142,6 +142,22 @@ type Retention struct {
 	ComplianceReports *int32 `json:"complianceReports"`
 }
 
+// LogStorageComponentName CRD enum
+type LogStorageComponentName string
+
+const (
+	ComponentNameECKOperator LogStorageComponentName = "ECKOperator"
+)
+
+// The ComponentResource struct associates a ResourceRequirements with a component by name
+type LogStorageComponentResource struct {
+	// ComponentName is an enum which identifies the component
+	// +kubebuilder:validation:Enum=ECKOperator
+	LogStorageComponentName LogStorageComponentName `json:"componentName"`
```

Comment:
I think we should just call this ComponentName, WDYT?
Either way this name and the comment above should match.

Reply:
Yeah good call, changed it to `ComponentName`.